plb/engine/controller/robot_controller.py

```diff
@@ -99,8 +100,7 @@ def parse_config(cls, cfgs: List[Union[CN, str]], primitiveController: Primitive
         rc = RobotsController()
         for eachOutCfg in outs:
             diffRobot = DiffRobot.load(eachOutCfg.path)
-            for shape in rc.append_robot(diffRobot, eachOutCfg.offset):
-                primitiveController.primitives.append(shape)
+            list(rc.append_robot(diffRobot, eachOutCfg.offset))
```

Comment:
I was initially confused by its yielding behavior since `for _ in rc.append_robot()` looks like its appending many robots but instead it's actually yielding the link primitives.

But then in this context, I do not need the yielded primitives so I was choosing between `list(rc.append_robot(...)` and ``for _ in rc.append_robot(): pass` and I found the former to be more succint.

Reply:
I find myself preferring a simple `rc.append_robot(...)` which optionally returns a list of primitives, but then the generator is not allowing me to do so. And since a robot do not have many primitives, the benefits a generator could bring seem minimal. 

Can you briefly explain the need for it?

Reply:
Oh previously the generator is used to append primitives created from a robot one by one. i.e.,
```py
for link_primitive in rc.append_robot(urdf_robot):
    primitives.append(link_primitive)
```

But currently with the help of the primitive registers, it seems that under no circumstance will we need to do so. You can just safely delete the yield or return a list of primitive instead. I would suggest the former as returning a list may utilize extra space.